simtool.h

```diff
@@ -702,6 +705,32 @@ class tool_stop_mover_t : public two_click_tool_t {
 	void read_start_position(player_t *player, const koord3d &pos);
 };
 
+class tool_change_city_of_building_t : public two_click_kartenboden_tool_t {
+private:
+	stadt_t* highlight_city = nullptr;
```

Comment:
Since the highlighted city should be always restorable from `default_param`, I wonder we should really use the state property here.

Reply:
Okay so after further investigation, the custom data & stadt_t pointer in the tool seems to be necessary for the desired working of the tool